src/main.ts

```diff
@@ -2,13 +2,26 @@ import 'source-map-support/register';
 import { logger } from './modules/common/Logger';
 import Stoa from './Stoa';
 
+import { ArgumentParser } from "argparse";
 import express from 'express';
 import { URL } from 'url';
 
-const port: number = Number(process.env.PORT) || 3836;
-const agora_address: URL = new URL(process.env.AGORA_ENDPOINT || "http://127.0.0.1:2826");
-// TODO: Replace hardcoded "127.0.0.1" with  agora_address.hostname (#106)
-const stoa: express.Application = new Stoa("database", "127.0.0.1", agora_address.port).stoa;
+// Parse the arguments
+const parser = new ArgumentParser();
+parser.add_argument('-g', '--agora', { help: "The endpoint of Agora" });
+parser.add_argument('-a', '--address', { help: "The address to which we bind to Stoa" });
+parser.add_argument('-p', '--port', { help: "The port on which we bind to Stoa" });
+parser.add_argument('-d', '--database', { help: "The file name of sqlite3 database" });
+let args = parser.parse_args();
 
-stoa.listen(port, () => logger.info(`Stoa API server listening at ${port}`))
+const address: string = args.address || "0.0.0.0";
+const port: number = Number(args.port || "3836");
+const agora_address: URL = new URL(args.agora || "http://127.0.0.1:2826");
+const database_filename: string = args.database || "database";
+logger.info(`Using Agora located at: ${agora_address.hostname}: ${agora_address.port}`);
+logger.info(`Listening to requests on: ${address}:${port}`);
```

Comment:
Ah this is already done at L26, so could you just replace the log message at L26 with this ?

Reply:
I fixed it